### DISCUSSION TITLE
Soft delete 정책 논의

### DISCUSSION THREAD

**Discussion by hectick** [woowacourse-teams/2023-edonymyeon#405] [백엔드 토론의 장]
현재 pr로 변경된 상황에서는, 
신고될 수 있는 항목인 post(+사진)과 comment(+사진), 그리고 여러 항목이 묶여있는 member가 soft delete 입니다.
이해를 돕기 위해 erd를 가져왔습니다. deleted 컬럼은 누런 형광펜을 쳐보았습니다.
<img width="981" alt="image" src="https://github.com/woowacourse-teams/2023-edonymyeon/assets/62106852/e5a97cf7-c1ba-4b14-a9d0-5e6bbca10284">

## 논의사항
1. 여기서 추가적으로 soft delete 해야할 테이블이 있을까요?
2. post_image_info와 comment_image_info가 soft delete 될 때, 실제 이미지 저장 디렉토리에 이미지는 남겨놓는 방향으로 진행할까요?
    - 신고를 위해 남겨놓는다 vs 저장 공간만 차지하니 지운다 vs 일단 남겨놓고 언젠간 지워주기..?

**Comment by This2sho** [comment]
1. `profile_image_info` 테이블도 멤버 수정 생기면 soft delete 적용해야할거 같습니다. 
  -> `ImageInfo`를 상속하는 모든 객체가 soft delete라 `ImageInfo`에 delete 상태 추가하는건 어떠신지?
  
 2. 일단 남겨놓고 언젠간 지워줘야할 거 같습니다링
